analyses/cell-type-wilms-tumor-06/scripts/05_copyKAT.R

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env Rscript
+
+# Run `copyKAT` for one sample with or without a healthy reference
+# copyKAT
+#
+# USAGE:
+# Rscript copyKAT.R \
+#   --sample_id SCPCS000194
+#   --ncore 16
+#
+
+library(optparse)
+library(Seurat)
+library(copykat)
+library(fs)
+# Parse arguments --------------------------------------------------------------
+# set up arguments
+option_list <- list(
+  make_option(
+    opt_str = c("-s", "--sample_id"),
+    type = "character",
+    default = "SCPCS000179",
+    help = "The sample_id of the sample to be used for inference of genomic copy number using copyKAT "
+  ),
+  make_option(
+    opt_str = c("-c", "--n_core"),
+    type = "integer",
+    default = 16,
+    help = "number of cores used to run copyKAT"
+  )
+  ,
+  make_option(
+    opt_str = c("-d", "--distance"),
+    type = "character",
+    default = "euclidean",
+    help = "method used to calculate distance in copyKAT"
+  )
+)
+
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# paths to data ----------------------------------------------------------------
+
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
+# Path to the result directory
+result_dir <- file.path(module_base, "results", opts$sample_id)
+
+
+# Create directories to save the results of copykat with/without reference using opts$distance
+dir.create(file.path(result_dir,  "05_copyKAT", "noref", opts$distance), recursive = TRUE)
+dir.create(file.path(result_dir,  "05_copyKAT", "ref", opts$distance), recursive = TRUE)
+
+# define scratch directory for tempory saving the output of copykat
+scratch_dir <- file.path(module_base, "scratch", opts$sample_id)
+dir.create(scratch_dir, recursive = TRUE)
+
+# path for copykat rds output
+name_ref <- glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, ".rds")
+name_ref_full <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, name_ref)
+name_no_ref <- glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, ".rds")
+name_no_ref_full <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, name_no_ref)
+
+# path to scratch and final heatmap file to copy over
+jpeg_file <- glue::glue(opts$sample_id,"_copykat_heatmap.jpeg")
+scratch_jpeg <- file.path(scratch_dir, jpeg_file)
+output_jpeg_ref <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, "_copykat_heatmap.jpeg"))
+output_jpeg_noref <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, "_copykat_heatmap.jpeg"))
```

Comment:
Do we _need_ to save these jpeg files? One reason I ask if that our pre-commit hooks currently are only allowing PNG images. Either way, unless the results are getting specifically used, you do not have to save each individual file that `copyKAT` outputs.

Reply:
I wanted to save the heatmaps to have then a quick look at the outputs with the different parameters (choice of normal and distance), as done for the [Ewing sarcoma analysis](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/analyses/cell-type-ewings/exploratory_analysis/03-copykat.Rmd) (lines 116 and 120)

What about [converting to png](https://stackoverflow.com/questions/44779804/converting-a-jpg-image-to-png) and then save the png?